ggml/include/ggml-opencl2.h

```diff
@@ -0,0 +1,39 @@
+// SPDX-FileCopyrightText: Copyright (c) Qualcomm Innovation Center, Inc. All rights reserved
+// SPDX-License-Identifier: MIT
+
+#ifndef GGML_OPENCL2_H
+#define GGML_OPENCL2_H
+
+#include "ggml.h"
+#include "ggml-backend.h"
+
+#ifdef  __cplusplus
+extern "C" {
+#endif
+
+#define CL_CHECK(err)                                               \
+    do {                                                            \
+        cl_int err_ = (err);                                        \
+        if (err_ != CL_SUCCESS) {                                   \
+            fprintf(stderr, "ggml_opencl: %s error %d at %s:%d\n",  \
+                #err, err_, __FILE__, __LINE__);                    \
+            GGML_ASSERT(0);                                                \
+        }                                                           \
+    } while (0)
```

Comment:
Can be moved in `ggml-opencl2.cpp`?

Reply:
Yep. Makes sense to keep the public header clean. Pushing an update in a minute.